src/sentry/services/hybrid_cloud/organization_actions/impl.py

```diff
@@ -0,0 +1,51 @@
+from typing import TypedDict
+
+from django.db import transaction
+from django.db.models.expressions import CombinedExpression
+
+from sentry.models import Organization, OrganizationStatus
+
+
+class OrganizationCreateOptions(TypedDict, total=False):
+    name: str
+    slug: str
+    status: OrganizationStatus
+    flags: CombinedExpression
+    default_role: int
+
+
+def create_organization_with_outbox_message(
+    *, create_options: OrganizationCreateOptions
+) -> Organization:
+    with transaction.atomic():
+        org: Organization = Organization.objects.create(**create_options)
+        Organization.outbox_for_update(org_id=org.id).save()
+    return org
+
+
+class OrganizationUpdateOptions(TypedDict, total=False):
+    name: str
+    status: OrganizationStatus
+    flags: CombinedExpression
+    default_role: int
```

Comment:
Don't organization updates also need to handle slugs?

Reply:
I was on the fence since our new push model will only change a slug in a single code path (Control silo outbox to region), so I was planning to rely on the upsert method to handle this case. It doesn't really make sense to exclude slug from here though so I'll go ahead and add it in to align the types on all of the methods now.